thrift.js

```diff
@@ -242,8 +251,16 @@ Thrift.prototype.resolve = function resolve(def) {
         return new ThriftSet(self.resolve(def.valueType), def.annotations);
     } else if (def.type === 'Map') {
         return new ThriftMap(self.resolve(def.keyType), self.resolve(def.valueType), def.annotations);
+    } else if (def.type === 'ServiceIdentifier') {
```

Comment:
It pains me to have code duplication here, but it does seem wise to track services separately from types. `claim` ensures that there are no name space collisions while allowing them to be tracked separately. :+1: